### ISSUE TITLE
Exception : InvalidVersion: Invalid version: '2.0.0-M4

### ISSUE THREAD

**Issue by KnowledgeSeekers** [Chaffelson/nipyapi#363]
* Nipyapi version: 0.2.8
* NiFi version: 2.0.0-M4
* NiFi-Registry version: 2.0.0-M4
* Python version: 3.12
* Operating System: Windows

### Description
I upgraded my Nifi from version 1.20 to 2.0.0-M4
I am trying to copy Nifi's content from version 1.20 of Nifi Registry 2.0.0-M4

### What I Did

Used Function : nipyapi.versioning.save_flow_ver 
Code 
```
pgdef=nipyapi.canvas.get_process_group(groupNameTGT, identifier_type='name')   
if (pgdef is None ):
    location = (location_x, location_y)
    root_process_group = nipyapi.canvas.get_process_group(rootPgIdTGT, 'id')
    pgdef=nipyapi.canvas.create_process_group(root_process_group, groupNameTGT, location, comment='TEST')
    savepg=nipyapi.versioning.save_flow_ver(pgdef, itemtregistriesTGT, bucketTGT, flow_name=workflowNameTGT, comment='create', desc='first', refresh=False)
```
Result :
`Exception : InvalidVersion: Invalid version: '2.0.0-M4'`

### Urgency

We can't use the package to migrate data.

**Comment by Chaffelson** [comment]
Hi - NiPy isn't currently tested against the 2.x release for NiFi.
It is something I plan on looking at, but not immediately.

**Comment by Chaffelson** [comment]
@KnowledgeSeekers you list your NiPyAPI version as 0.2.8, but the current version is 0.20.0.
Can you please confirm you are using a current version?

**Comment by KnowledgeSeekers** [comment]
Besides, there is a small issue when I use the last version : 

import nipyapi
print(nipyapi.__version__)
**0.19.1**

There might be a bug here : 
![image](https://github.com/user-attachments/assets/f42c01ef-98c9-4ed4-9797-7b9177d62ba0)

**Comment by Chaffelson** [comment]
> Besides, there is a small issue when I use the last version :
Yes, good spotting. Bumpversion didn't catch it in that release and neither did I, I've fixed it for the upcoming 0.21 release.

I've found and fixed the version test issue, but there are still a lot of compatibility issues to support NiFi 2.x without a large refactor. I'm seeing if I can do a minimal set of changes to allow it to function and will let you know.

**Comment by KnowledgeSeekers** [comment]
Thx for the information. Any idea about the release date of compatible version with Nifi 2 ?

**Comment by Chaffelson** [comment]
> Thx for the information. Any idea about the release date of compatible version with Nifi 2 ?

Roughly nowish
https://github.com/Chaffelson/nipyapi/pull/368
😃

**Comment by Chaffelson** [comment]
Fixed in [v0.21.0](https://github.com/Chaffelson/nipyapi/releases/tag/v0.21.0) release
